src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2007,6 +2008,20 @@ internal PointerTypeSymbol CreatePointerTypeSymbol(TypeSymbol elementType, Nulla
             throw new NotImplementedException();
         }
 
+        private ImmutableQueue<SourceMethodSymbolWithAttributes> _moduleInitializerMethods = ImmutableQueue<SourceMethodSymbolWithAttributes>.Empty;
+
+        internal void AddModuleInitializerMethod(SourceMethodSymbolWithAttributes method)
+        {
+            Debug.Assert(!_declarationDiagnosticsFrozen);
+            ImmutableInterlocked.Enqueue(ref _moduleInitializerMethods, method);
+        }
+
+        internal ImmutableQueue<SourceMethodSymbolWithAttributes> GetModuleInitializerMethods()
```

Comment:
>internal [](start = 8, length = 8)

private? BTW, it is not clear why do we need this method, we can access the field directly in GenerateModuleInitializer?

Reply:
If we end up freezing the collection when this is called, I like the separation from the consuming method. Can I keep it as a private method?

Reply:
> Can I keep it as a private method?

That would be Ok, I guess. But having  a method to return ```ConcurrentSet<MethodSymbol>```, which is mutable feels like a code smell to me. There is only one consumer, it is the same class and we can assert ```_declarationDiagnosticsFrozen``` there.

---
In reply to: [414530636](https://github.com/dotnet/roslyn/pull/43281#discussion_r414530636) [](ancestors = 414530636)

Reply:
Ah, since it ends up being a mutable collection it does feel wrong.